app/api/probe_dock/metrics_api.rb

```diff
@@ -62,6 +68,12 @@ def apply_projects_and_users_filters(rel_for_filtering)
           rel
         end
 
+        def apply_projects_and_users_filters(rel_for_filtering)
+          rel = apply_projects_filters rel_for_filtering
```

Comment:
In previous methods, you have defined a `rel_for_filtering` argument and the first thing you do is re-declare a `rel` variable before transforming it (`rel = rel_for_filtering`). In this method, you transform it directly on the first call, and this line and the next have different argument names, making it in my opinion less clear. Also, the ruby style guide recommends to "avoid shadowing methods with local variables unless they are both equivalent" (we'll have some refactoring to do there).

Personally, I think that if a method is called `apply_some_filter` and takes one argument, it's superfluous to call that argument `something_for_filtering` when you could just call it `something`. It obvious that it's for filtering. 

I think this format would be clearer:

```
def apply_some_filter rel
  filtered_rel = rel

  # Filter by ...
  filtered_rel = rel.where(...)

  # Group by ... if ...
  if some_condition
    filtered_rel = filtered_rel.group(...)
  end

  filtered_rel
end
```

And finally, I would call the two previous methods `apply_some_filter` instead of `apply_some_filters` (remove the "s" at the end). There's multiple users and projects, but each of those methods applies one filter. This of course doesn't apply to this method.